dbms/src/Debug/MockExecutor/WindowBinder.cpp

```diff
@@ -121,25 +121,35 @@ bool WindowBinder::toTiPBExecutor(tipb::Executor * tipb_executor, int32_t collat
         if (frame.start.has_value())
         {
             auto * start = mut_frame->mutable_start();
-            start->set_offset(std::get<2>(frame.start.value()));
-            start->set_unbounded(std::get<1>(frame.start.value()));
-            start->set_type(std::get<0>(frame.start.value()));
+            start->set_offset(frame.start->getOffset());
+            start->set_unbounded(frame.start->isUnbounded());
+            start->set_type(frame.start->getBoundType());
+            if (frame.start->isRangeFrame())
+            {
+                start->set_allocated_frame_range(frame.start->robRangeFrame());
+                start->set_cmp_data_type(frame.start->getCmpDataType());
+            }
         }
 
         if (frame.end.has_value())
         {
             auto * end = mut_frame->mutable_end();
-            end->set_offset(std::get<2>(frame.end.value()));
-            end->set_unbounded(std::get<1>(frame.end.value()));
-            end->set_type(std::get<0>(frame.end.value()));
+            end->set_offset(frame.end->getOffset());
+            end->set_unbounded(frame.end->isUnbounded());
+            end->set_type(frame.end->getBoundType());
+            if (frame.end->isRangeFrame())
+            {
+                end->set_allocated_frame_range(frame.end->robRangeFrame());
+                end->set_cmp_data_type(frame.end->getCmpDataType());
+            }
         }
     }
 
     auto * children_executor = window->mutable_child();
     return children[0]->toTiPBExecutor(children_executor, collator_id, mpp_info, context);
 }
 
-ExecutorBinderPtr compileWindow(ExecutorBinderPtr input, size_t & executor_index, ASTPtr func_desc_list, ASTPtr partition_by_expr_list, ASTPtr order_by_expr_list, mock::MockWindowFrame frame, uint64_t fine_grained_shuffle_stream_count)
+ExecutorBinderPtr compileWindow(ExecutorBinderPtr input, size_t & executor_index, ASTPtr func_desc_list, ASTPtr partition_by_expr_list, ASTPtr order_by_expr_list, mock::MockWindowFrame & frame, uint64_t fine_grained_shuffle_stream_count)
```

Comment:
```suggestion
ExecutorBinderPtr compileWindow(ExecutorBinderPtr input, size_t & executor_index, ASTPtr func_desc_list, ASTPtr partition_by_expr_list, ASTPtr order_by_expr_list, mock::MockWindowFrame frame, uint64_t fine_grained_shuffle_stream_count)
```
It looks like copy is fine

Reply:
okk